app.js

```diff
@@ -9,65 +9,141 @@ const reportEl = document.getElementById('report');
     // increment the middle change count state
-    
+    middleCount++;
     // update the dom for the middle (NOTE: use style.backgroundImage on the middleEl div instead of trying to set the .src -- it's NOT an img tag!)
-
+    middleEl.style.backgroundImage = `url("./assets/${value}-middle.png")`;
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
 
 bottomDropdown.addEventListener('change', () => {
     // get the value of the bottom dropdown
-
+    const value = bottomDropdown.value;
     // increment the bottom change count state
-    
+    bottomCount++;
     // update the dom for the bottom (NOTE use style.backgroundImage on the bottomEl div instead of trying to set the .src -- it's NOT an img tag!)
-
+    bottomEl.style.backgroundImage = `url("./assets/${value}-pants.png")`;
     // update the stats to show the new count (call displayStats() to do this work)
+    displayStats();
 });
 
 catchphraseButton.addEventListener('click', () => {
     // get the value of the catchphrase input
-    
+    const value = catchphraseInput.value;
     // push the new catchphrase to the catchphrase array in state
-
+    catchPhrases.push(value);
     // clear out the form input's value so it's empty to the user
-   
+    catchphraseInput.value = '';
     // update the dom to show the new catchphrases (refactor to/call displayCatchphrases to do this work)
+    displayCatchphrases();
+    //hide the button again so you cannot move forward without inputting
+    catchphraseButton.classList.add('hidden');
+});
 
+catchphraseInput.addEventListener('input', () => {
+    catchphraseButton.classList.remove('hidden');
+});
+
+nameButton.addEventListener('click', () => {
+    const value = nameInput.value;
+    characterNames.push(value);
+    nameInput.value = '';
+    displayName();
 });
 
 function displayStats() {
     // text content of the reportEl to tell the user how many times they've changed each piece of the state
+    reportEl.textContent = `You have changed the head ${headCount} times!
+    you have also changed the middle ${middleCount} times! Finally, you have
+    changed the bottom ${bottomCount} times!`;
 }
 
 function displayCatchphrases() {
     // clear out the DOM for the currently displayed catchphrases
-
+    catchphrasesEl.textContent = '';
     // loop through each catchphrase in state
-   
-    // and for each catchphrase
-    
-    // create an HTML element with the catchphrase as its text content
-    
+    for (let catchPhrase of catchPhrases) {
+    // for each catchphrase create an HTML element with the catchphrase as its text content
+        const pTag = document.createElement('p');
     // and append that HTML element to the cleared-out DOM
+        catchphrasesEl.append(pTag);
+
+        pTag.textContent = catchPhrase;
+        pTag.classList.add('catchphraseAdded');
+    }
+    if (catchPhrases.length >= 4) {
+        catchPhrases.shift();
```

Comment:
nice touch with the conditional .shift! love seeing you research and add features like this to your app.